test/extensions/filters/network/zookeeper_proxy/config_test.cc

```diff
@@ -60,6 +145,139 @@ stat_prefix: test_prefix
+  const std::string yaml = R"EOF(
+stat_prefix: test_prefix
+default_latency_threshold: "0.15s"
+  )EOF";
+
+  ZooKeeperProxyProtoConfig proto_config;
+  TestUtility::loadFromYamlAndValidate(yaml, proto_config);
+
+  testing::NiceMock<Server::Configuration::MockFactoryContext> context;
+  ZooKeeperConfigFactory factory;
+
+  Network::FilterFactoryCb cb = factory.createFilterFactoryFromProto(proto_config, context);
+  Network::MockConnection connection;
+  EXPECT_CALL(connection, addFilter(_));
+  cb(connection);
+}
+
+TEST(ZookeeperFilterConfigTest, ConfigWithConnectLatencyThreshold) {
+  const std::string yaml = R"EOF(
+stat_prefix: test_prefix
+latency_threshold_overrides:
+  - opcode: Connect
+    threshold: "0.151s"
+  )EOF";
+
+  ZooKeeperProxyProtoConfig proto_config;
+  TestUtility::loadFromYamlAndValidate(yaml, proto_config);
+
+  testing::NiceMock<Server::Configuration::MockFactoryContext> context;
+  ZooKeeperConfigFactory factory;
+
+  Network::FilterFactoryCb cb = factory.createFilterFactoryFromProto(proto_config, context);
+  Network::MockConnection connection;
+  EXPECT_CALL(connection, addFilter(_));
+  cb(connection);
+}
+
+TEST(ZookeeperFilterConfigTest, FullConfig) {
+  const std::string yaml = R"EOF(
+stat_prefix: test_prefix
+max_packet_bytes: 1048576
+default_latency_threshold: "0.1s"
+latency_threshold_overrides:
+  - opcode: Connect
+    threshold: "0.151s"
+  - opcode: Create
+    threshold: "0.152s"
+  - opcode: Delete
+    threshold: "0.153s"
+  - opcode: Exists
+    threshold:
+      nanos: 154000000
+  - opcode: GetData
+    threshold:
+      nanos: 155000000
+  - opcode: SetData
+    threshold:
+      nanos: 156000000
+  - opcode: GetAcl
+    threshold:
+      nanos: 157000000
+  - opcode: SetAcl
+    threshold:
+      nanos: 158000000
+  - opcode: GetChildren
+    threshold:
+      nanos: 159000000
+  - opcode: Sync
+    threshold:
+      nanos: 160000000
+  - opcode: Ping
+    threshold:
+      nanos: 161000000
+  - opcode: GetChildren2
+    threshold:
+      nanos: 162000000
+  - opcode: Check
+    threshold:
+      nanos: 163000000
+  - opcode: Multi
+    threshold:
+      nanos: 164000000
+  - opcode: Create2
+    threshold:
+      nanos: 165000000
+  - opcode: Reconfig
+    threshold:
+      nanos: 166000000
+  - opcode: CheckWatches
+    threshold:
+      nanos: 167000000
+  - opcode: RemoveWatches
+    threshold:
+      nanos: 168000000
+  - opcode: CreateContainer
+    threshold:
+      nanos: 169000000
+  - opcode: CreateTtl
+    threshold:
+      nanos: 170000000
+  - opcode: Close
+    threshold:
+      nanos: 171000000
+  - opcode: SetAuth
+    threshold:
+      nanos: 172000000
+  - opcode: SetWatches
+    threshold:
+      nanos: 173000000
+  - opcode: GetEphemerals
+    threshold:
+      nanos: 174000000
+  - opcode: GetAllChildrenNumber
+    threshold:
+      nanos: 175000000
+  - opcode: SetWatches2
+    threshold:
+      nanos: 176000000
+  )EOF";
```

Comment:
can we refactor this test so we can check that these configs actually make it through?

I think you can do this deeper test much more tersely than even what you have here, which IIUC is only verifying that you don't throw.

You can hake a test-fixture helper method to formulate a coherent config fragment with just one setting, and then verify that setting made it through correctly.